### DISCUSSION TITLE
[Typescript] GeojsonLayer error on FeatureCollection type input

### DISCUSSION THREAD

**Discussion by JRVID** [visgl/deck.gl#7276] [Q&A]
Hey guys - another type issue i've found along with https://github.com/visgl/deck.gl/discussions/7258 from the new types.

I can see that the geoJsonLayerProps uses the Feature type... this implies that the type of data that the layer can render is only a single geoJson when this is not the case as I commonly pass a FeatureCollection into the layer to render multiple polygons.

Is there a way around this without having to do a Ts-ignore on the layer currently? The 3rd party danmarshall typings actually never solved this issue to my knowledge either

**Comment by lstuartfry** [comment]
+1 on this discussion - I'm also struggling somewhat with typing the parameters returned from any of the `Accessor`-type functions you can pass to a `Layer` class. For example, typing the parameters for something like the `getTextSize` function, which expects `object` and `objectInfo` params, according to [the Accessor docs](https://deck.gl/docs/developer-guide/using-layers#accessors)

**Comment by krishamehta903** [comment]
> Hey guys - another type issue i've found along with #7258 from the new types.
> 
> I can see that the geoJsonLayerProps uses the Feature type... this implies that the type of data that the layer can render is only a single geoJson when this is not the case as I commonly pass a FeatureCollection into the layer to render multiple polygons.
> 
> Is there a way around this without having to do a Ts-ignore on the layer currently? The 3rd party danmarshall typings actually never solved this issue to my knowledge either

I'm having similar experiences and created an issue on Github fyi - https://github.com/visgl/deck.gl/issues/7571
